src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -110,7 +110,7 @@ namespace AirLoopHVACDOAS {
 
         if (this->SumMassFlowRate == 0.0 && !state.dataGlobal->BeginEnvrnFlag) {
             state.dataLoopNodes->Node(this->m_CompPointerAirLoopMixer->OutletNodeNum).MassFlowRate = 0.0;
-            return;
+            if (state.dataGlobal->KickOffSimulation) return;
         }
 
         this->CalcAirLoopDOAS(state, FirstHVACIteration);
```

Comment:
This fix allows to execute *CalcAirLoopDOAS* function even if the *SumMassFlowRate* is zero and the *KickOffSimulation* is set to *FALSE*. Previously, the function  *CalcAirLoopDOAS* is not executed if the supply air flow rate is zero. Thereby, report variables for any component on the DOAS system will not be reset to zero, instead values of previous time step is carried over until the system flow rate is > 0.

Reply:
My first attempt was to removed the return statement altogether but the simulation fatal out because of SetPointManager has not yet set the controlled nodes set-point values. The `KickOffSimulation` flag allows to circumvent that situation. Is there any other way to do the same?

Reply:
Not sure. I see `KickOffSimulation` checks scattered throughout the code, but nearly all of them as related for fault adjustments. Leave this as-is and I'll try to figure out why other systems don't fail during kickoff but this one does.

Reply:
I reproduced the severe error message I was getting when I commented out the `return` statement. See below:

```
   ** Severe  ** HVACControllers: Missing temperature setpoint for controller type=Controller:WaterCoil Name="OA HC CONTROLLER 1"
   **   ~~~   ** Node Referenced (by Controller)=OA HEATING COIL 1 AIR OUTLET NODE
   **   ~~~   **   use a Setpoint Manager with Control Variable = "Temperature" to establish a setpoint at the controller sensed node.
   ** Severe  ** HVACControllers: Missing temperature setpoint for controller type=Controller:WaterCoil Name="OA CC CONTROLLER 1"
   **   ~~~   ** Node Referenced (by Controller)=AIRLOOPDOASSPLITTERINLET
   **   ~~~   **   use a Setpoint Manager with Control Variable = "Temperature" to establish a setpoint at the controller sensed node.
```

Reply:
`ManageSetpoints` is called here [in `SimHVAC`](https://github.com/NREL/EnergyPlus/blob/797ffe32e8141c6c822db9f26ac62b7ed1603302/src/EnergyPlus/HVACManager.cc#L799-L821) before `SimSelectedEquipment` which ultimately calls `SimAirLoopHVACDOAS`. So it seems `ManageSetpoints` should have already initialized those setpoints.

Reply:
Found it!! The DOAS is copying entire node structs [here](https://github.com/NREL/EnergyPlus/blob/797ffe32e8141c6c822db9f26ac62b7ed1603302/src/EnergyPlus/AirLoopHVACDOAS.cc#L944-L948) which overwrites the setpoints. I think this `if (this->SumMassFlowRate == 0.0) {` block can be deleted now. Are there any other places that are checking for `SumMassFlowRate == 0`?

Reply:
@mjwitte Great!. There are only two places. The one you found out in `initAirLoopDOAS(state, FirstHVACIteration)` and the offending code for this issue in `AirLoopDOAS::SimAirLoopHVACDOAS` [here](https://github.com/NREL/EnergyPlus/blob/182064950_Issue9342/src/EnergyPlus/AirLoopHVACDOAS.cc#L111-L114).